time stamp
bool[] connections
string[] ids
uint32 length
